diff --git a/cpplint.orig.py b/cpplint.py
index 86c1027..543a609 100644
--- a/cpplint.orig.py
+++ b/cpplint.py
@@ -57,6 +57,7 @@ import sys
 import sysconfig
 import unicodedata
 import xml.etree.ElementTree
+import fnmatch
 
 # if empty, use defaults
 _valid_extensions: set[str] = set()
@@ -1256,10 +1257,10 @@ class _IncludeState:
     # needs to move backwards, CheckNextIncludeOrder will raise an error.
     _INITIAL_SECTION = 0
     _MY_H_SECTION = 1
-    _C_SECTION = 2
-    _CPP_SECTION = 3
-    _OTHER_SYS_SECTION = 4
-    _OTHER_H_SECTION = 5
+    _OTHER_SYS_SECTION = 2
+    _OTHER_H_SECTION = 3
+    _C_SECTION = 4
+    _CPP_SECTION = 5
 
     _TYPE_NAMES = {
         _C_SYS_HEADER: "C system header",
@@ -4125,13 +4126,14 @@ def CheckSpacingForFunctionCall(filename, clean_lines, linenum, error):
             # If the closing parenthesis is preceded by only whitespaces,
             # try to give a more descriptive error message.
             if re.search(r"^\s+\)", fncall):
-                error(
-                    filename,
-                    linenum,
-                    "whitespace/parens",
-                    2,
-                    "Closing ) should be moved to the previous line",
-                )
+                pass
+                # error(
+                #     filename,
+                #     linenum,
+                #     "whitespace/parens",
+                #     2,
+                #     "Closing ) should be moved to the previous line",
+                # )
             else:
                 error(filename, linenum, "whitespace/parens", 2, "Extra space before )")
 
@@ -5878,7 +5880,7 @@ def CheckIncludeLine(filename, clean_lines, linenum, include_state, error):
     #
     # We also make an exception for Lua headers, which follow google
     # naming convention but not the include convention.
-    match = re.match(r'#include\s*"([^/]+\.(.*))"', line)
+    match = False#re.match(r'#include\s*"([^/]+\.(.*))"', line)
     if (
         match
         and IsHeaderExtension(match.group(2))
@@ -5926,8 +5928,9 @@ def CheckIncludeLine(filename, clean_lines, linenum, include_state, error):
         # We DO want to include a 3rd party looking header if it matches the
         # filename. Otherwise we get an erroneous error "...should include its
         # header" error later.
-        third_src_header = False
+        third_src_header = True#False
         for ext in GetHeaderExtensions():
+            break # minimize diff
             basefilename = filename[0 : len(filename) - len(fileinfo.Extension())]
             headerfile = basefilename + "." + ext
             headername = FileInfo(headerfile).RepositoryName()
@@ -7633,6 +7636,7 @@ def ProcessConfigOverrides(filename):
 
         try:
             with open(cfg_file, encoding="utf8", errors="replace") as file_handle:
+                skip = False
                 for line in file_handle:
                     line, _, _ = line.partition("#")  # Remove comments.
                     if not line.strip():
@@ -7641,6 +7645,16 @@ def ProcessConfigOverrides(filename):
                     name, _, val = line.partition("=")
                     name = name.strip()
                     val = val.strip()
+
+                    # will skip everything below if file doesn't match the pattern
+                    if name == 'filematch':
+                        skip = not fnmatch.fnmatch(os.path.basename(filename), val)
+                        continue
+
+                    # skip everything until there's a new 'filematch'
+                    if skip:
+                        continue
+
                     if name == "set noparent":
                         keep_looking = False
                     elif name == "filter":
